java/core/src/main/java/com/lancedb/lance/Version.java

```diff
@@ -0,0 +1,62 @@
+/*
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+package com.lancedb.lance;
+
+import java.time.ZonedDateTime;
+import java.util.TreeMap;
+
+public class Version {
+  private final long id;
+  private final ZonedDateTime dataTime;
+  private final TreeMap<String, String> metadata;
```

Comment:
why ping to TreeMap instead of just Map?

Reply:
The underlying rust code here is declared as BTreeMap.

```
#[derive(Deserialize, Serialize)]
pub struct Version {
    /// version number
    pub version: u64,

    /// Timestamp of dataset creation in UTC.
    pub timestamp: DateTime<Utc>,

    /// Key-value pairs of metadata.
    pub metadata: BTreeMap<String, String>,
}
```
I thought this was a explicit sorting declaration

Reply:
maybe SortedMap then? At least we should not expose a specific implementation in Java interfaces. We can use tree map to assign value that is totally fine.

Reply:
Absolutely